### PR TITLE
Codify lodash usage in eslint

### PR DESCRIPTION
**Pull request by Janpot**
Avoid kitchensink libraries, especially unmaintained ones like `lodash`. We write a slightly more verbose, but a universally understood style of javascript.

### CODE REVIEW

File: packages/toolpad-app/src/runtime/ToolpadApp.tsx [hunk 1]
```diff
@@ -34,6 +34,8 @@ import {
   JsExpressionAttrValue,
 } from '@mui/toolpad-core';
 import { createProvidedContext, useAssertedContext } from '@mui/toolpad-utils/react';
+import { filterKeys, mapProperties, mapValues } from '@mui/toolpad-utils/collections';
+import { set as setObjectPath } from 'lodash-es';
```
Comment (apedroferreira) [inline packages/toolpad-app/src/runtime/ToolpadApp.tsx:38]: If we plan to eventually completely get rid of `lodash` we could maybe create utility methods for these operations, which for now would still use `lodash` but then we would know where to replace it.
Just a possibility, maybe it helps to have the `lodash` imports less spread out.

**Review by apedroferreira — APPROVED** [review]
Sounds good, should we create an issue to replace `groupBy` and `isEqual`?
I remember also using `setWith` for a specific case in bindings, that one might be tougher to replace but still possible.

File: packages/toolpad-app/src/runtime/ToolpadApp.tsx [hunk 2]
```diff
@@ -1226,7 +1228,7 @@ function QueryNode({ page, node }: QueryNodeProps) {
     Object.fromEntries(node.params ?? []),
   );
 
-  const configBindings = _.pick(node.attributes, USE_DATA_QUERY_CONFIG_KEYS);
+  const configBindings = filterKeys(node.attributes, (key) => USE_DATA_QUERY_CONFIG_KEYS.has(key));
```
Comment (apedroferreira) [inline packages/toolpad-app/src/runtime/ToolpadApp.tsx:1231]: We could abstract this to its our own `pick` function maybe?

Comment (Janpot) [reply packages/toolpad-app/src/runtime/ToolpadApp.tsx]: Removing it altogether

File: packages/toolpad-app/src/runtime/ToolpadApp.tsx (hunk 1 above)
Comment (Janpot) [reply packages/toolpad-app/src/runtime/ToolpadApp.tsx:38]: To be honest, I'm not sure we should even have utility functions for this.

Comment (apedroferreira) [reply packages/toolpad-app/src/runtime/ToolpadApp.tsx:38]: Got it, yeah for this specific type of operation maybe just normal Javascript would be possible and be more type-safe.

**Pull request merged**